tests/integration/high_order_matches/test_document.py

```diff
@@ -0,0 +1,63 @@
+import os
+import shutil
+
+import pytest
+
+from jina.flow import Flow
+from jina import Document, Executor, requests
+from tests import validate_callback
+
+
+def validate(req):
+    assert len(req.docs) == 1
+    assert len(req.docs[0].matches) == 5
+    assert len(req.docs[0].matches[0].matches) == 5
+    assert len(req.docs[0].matches[-1].matches) == 5
+    assert len(req.docs[0].matches[0].matches[0].matches) == 0
+
+
+class MyExecutor(Executor):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.doc = self.requests
+
+    @requests(on='index')
+    def index(self, docs, **kwargs):
+        self.doc = docs[0]
+        for match in self.doc.matches:
+            for i in range(5):
+                match.matches.append(Document())
+
+        return self.doc
+
+
+def test_high_order_matches(mocker):
```

Comment:
Good job, lets keep only the second test since it includes also the first

Reply:
Alright